File: graphify/tests/fixtures/sample.kt
import kotlinx.coroutines.delay
import kotlin.math.max

data class Config(val baseUrl: String, val timeout: Int)

class HttpClient(private val config: Config) {
    fun get(path: String): String {
        return buildRequest("GET", path)
    }

    fun post(path: String, body: String): String {
        return buildRequest("POST", path)
    }

    private fun buildRequest(method: String, path: String): String {
        return "$method ${config.baseUrl}$path"
    }
}

fun createClient(baseUrl: String): HttpClient {
    val config = Config(baseUrl, 30)
    return HttpClient(config)
}

File: graphify/tests/fixtures/sample.scala
import scala.collection.mutable.ListBuffer

case class Config(baseUrl: String, timeout: Int)

class HttpClient(config: Config) {
  def get(path: String): String = {
    buildRequest("GET", path)
  }

  def post(path: String, body: String): String = {
    buildRequest("POST", path)
  }

  private def buildRequest(method: String, path: String): String = {
    s"$method ${config.baseUrl}$path"
  }
}

object HttpClientFactory {
  def create(baseUrl: String): HttpClient = {
    new HttpClient(Config(baseUrl, 30))
  }
}
